anyway/widgets/widget_utils.py

```diff
@@ -123,3 +125,30 @@ def format_2_level_items(
             series_data.append({BE_CONST.LKEY: l2, BE_CONST.VAL: num})
         res.append({BE_CONST.LKEY: l1, BE_CONST.SERIES: series_data})
     return res
+
+
+def second_level_fill_and_sort(data: dict, default_order: dict) -> dict:
+    for num, value in data.items():
+        new_value = copy.deepcopy(default_order)
+        for key, value_in in value.items():
+            new_value[key] += value_in
+        data[num] = new_value
+
+    return data
+
+
+def fill_and_sort_by_numeric_range(
+    data: defaultdict, numeric_range: typing.Iterable, default_order: dict
+) -> Dict[int, dict]:
+    for item in numeric_range:
+        if item not in data:
+            data[item] = default_order
+    return dict(sorted(data.items()))

```

Comment:
not sure what is the purpose of sorting - dict is always unsorted

Reply:
From [Python version 3.7 dict are preserving order](https://docs.python.org/3.7/whatsnew/3.7.html) :
![image](https://user-images.githubusercontent.com/58947331/150652853-94eeac24-f63b-48b0-989e-31c4e6cdfbe3.png)

Our image is based on Ubuntu 20.04 which default to Python 3.8.2 so the dict will preserve order

Reply:
wow cool, didn't know that